### ISSUE TITLE
"Fancy" PS1 fails to eval under MSYS2 and on OSX

### ISSUE THREAD

**Issue by LankyCyril** [LankyCyril/Dotfiles#13]
However, it works just fine in Cygwin, Ubuntu 14.04 (and presumably Debian Wheezy).

```
bash: command substitution: line 9: syntax error near unexpected token `)'
bash: command substitution: line 9: `   }')'
bash: command substitution: line 1: syntax error near unexpected token `)'
bash: command substitution: line 1: `[ "0" = "0" ] && echo "" || echo " + $((0))")'
```

**Comment by LankyCyril** [comment]
Potentially related to these issues:
https://sourceforge.net/p/msys2/tickets/17/
http://unix.stackexchange.com/questions/28369/linux-from-scratchs-bash-problem-syntax-error

Maybe bash is not compiled with full Yacc support.
**Consider** either recompiling it (and then patching it on every clean install of MSYS2) or finally biting the bullet and moving to `zsh`.

**Comment by LankyCyril** [comment]
Fixed in commit `ea27351`, thanks to an explanation at http://stackoverflow.com/a/21561763.
